app/controllers/repositories_controller.rb

```diff
@@ -23,6 +25,28 @@ def toggle_star
     render template: "repositories/star", locals: { user: current_user }
   end
 
+  # Removes all the tags that belong to this repository, and removes it.
+  def destroy
+    # First of all we mark the repo and the tags, so we don't have concurrency
+    # problems when "delete" events come in.
+    @repository.tags.update_all(marked: true)
+    @repository.update_attributes(marked: true)
+
+    # Remove all tags, effectively removing them from the registry too.
+    @repository.groupped_tags.map { |t| t.first.delete_by_digest!(current_user) }
+
+    # Delete this repository if all tags were successfully deleted.
+    if @repository.reload.tags.any?
+      redirect_to repository_path(@repository), alert: "Could not remove all the tags"
```

Comment:
It is logged, because it ends up calling `Tag#delete_by_digest!`, which logs any errors. Maybe I can provide an extra line of: "Error: tags [...] could not be removed" (the reason on why they have been removed is logged by `Tag#delete_by_digest!`)